### PR TITLE
Fix all 404 links

### LINKED ISSUE #228: Fix all broken links
Following the replatform, some links now require an additional `/docs/` prepended, as that is a new node in our content tree. Further, there are a few leftover links from before the content reorganization. Finally, some links are out-of-date or have been broken for some time.

All links should lead somewhere and resolve to a non-400 code.

Thanks to @olivekl for pointing out a number of broken links in #43.

### PR DESCRIPTION
**Pull request by smythp**
All links on the docs site that resolved to 404 now lead somewhere

Some links changed to more useful or up-to-date links

Added link to Slack invite

References #43

Resolves #228

### CODE REVIEW

**Comment by normal-coder** [comment]
# Note: This is strongly not recommended! 

# Please try to solve this problem by Hugo engine.

@smythp @ltagliaferri

This is equivalent to tampering with all your routes once. This will cause all existing Sigstore materials and external links to be unable to get the correct jump. (In fact, this is a bug left over from migrating Hugo)

In addition, if the existing document address is changed, it does not look elegant. For a docs second-level domain, there is still a docs on the route, except that it brings extra operational burden to people like `www`, what's the point?

```diff
+ https://docs.sigstore.dev/docs/about/tooling/
- https://docs.sigstore.dev/about/tooling/
```

**Comment by smythp** [comment]
In principle, this is absolutely correct. Moving all of the URLs down to docs will break most external links to the site.

However, I'll note that the site has undergone a major content reorganization as of last week, and among those changes is the renaming of (I think) every resource category. For example, many of the resources that were once under "cosign" are now under "signing." When the Hugo migration was undertaken, only perhaps 4-5 links had the same URLs (about half of the items now under "about"). About 20% of the link fixes in this PR are actually internal links not updated during the content migration.

If more than a handful of links hadn't been reorganized, I agree that it would make sense to invest the effort to remove the "docs" level from the hierarchy. If there's interest in this expressed here, I can undertake to do that. However, I'll note that the Doks theme does hard-code behavior based on the presence of a docs folder, so this effort is more like creating our own theme, in terms of upfront effort and ongoing maintenance.

Thanks for this comment. I'm sympathetic to this as someone who has worked in libraries and favors backward compatibility. I'm curious to know what other thoughts there are on this issue, or what you think of the above.

tl;dr Most URLs have already changed before the Hugo migration, maybe 95%.

**Comment by normal-coder** [comment]
> 原则上来说，这是绝对正确的。将所有 URL 移至文档将破坏该网站的大多数外部链接。
> 
> 不过，我要指出的是，截至上周，该网站已经进行了重大内容重组，其中包括对（我认为）每个资源类别的重命名。例如，许多曾经处于“共同签名”状态的资源现在处于“签名”状态。当进行 Hugo 迁移时，大约只有 4-5 个链接具有相同的 URL（现在大约有一半的项目位于“about”下）。此 PR 中大约 20% 的链接修复实际上是内容迁移期间未更新的内部链接。
> 
> 如果没有重新组织多个链接，我同意投入精力从层次结构中删除“文档”级别是有意义的。如果有人对此感兴趣，我可以承诺这样做。不过，我会注意到，Doks 主题根据文档文件夹的存在进行硬编码行为，因此就前期工作和持续维护而言，这项工作更像是创建我们自己的主题。
> 
> 感谢您的评论。作为一个在图书馆工作过并支持向后兼容性的人，我对此表示同情。我很想知道关于这个问题还有什么其他想法，或者您对上述内容有何看法。
> 
> tl;dr 大多数 URL 在 Hugo 迁移之前已经发生了变化，也许 95%。



> In principle, this is absolutely correct. Moving all of the URLs down to docs will break most external links to the site.
> 
> However, I'll note that the site has undergone a major content reorganization as of last week, and among those changes is the renaming of (I think) every resource category. For example, many of the resources that were once under "cosign" are now under "signing." When the Hugo migration was undertaken, only perhaps 4-5 links had the same URLs (about half of the items now under "about"). About 20% of the link fixes in this PR are actually internal links not updated during the content migration.
> 
> If more than a handful of links hadn't been reorganized, I agree that it would make sense to invest the effort to remove the "docs" level from the hierarchy. If there's interest in this expressed here, I can undertake to do that. However, I'll note that the Doks theme does hard-code behavior based on the presence of a docs folder, so this effort is more like creating our own theme, in terms of upfront effort and ongoing maintenance.
> 
> Thanks for this comment. I'm sympathetic to this as someone who has worked in libraries and favors backward compatibility. I'm curious to know what other thoughts there are on this issue, or what you think of the above.
> 
> tl;dr Most URLs have already changed before the Hugo migration, maybe 95%.

Maybe you can consider trying Slug?

**Comment by ltagliaferri** [comment]
@normal-coder — we addressed this in a [recent info architecture change](https://github.com/sigstore/docs/pull/210/files#diff-fd40fde62534b781806d746f82ba621d39c640f85029c1524582d4b137701cfe), but the extensive re-platform lost some of these files which we should bring back, I agree! I created an issue to track: https://github.com/sigstore/docs/issues/230

**Comment by ltagliaferri** [comment]
I renamed with "WIP" for "work in progress" @smythp if you can change the flag when this is ready for review. Thanks!

**Comment by smythp** [comment]
OK, fixed the links and linted the pages I touched. In theory, there should be no current broken links on the site after this is merged.

**Review by ltagliaferri — APPROVED** [review]
